### DISCUSSION TITLE
Argument of type ReadableStream<Uint8Array> is not assignable to parameter of type ReadableStream<any>

### DISCUSSION THREAD

**Discussion by timothymalcham** [remix-run/remix#8522] [Q&A]
I'm using a node lib (csv-batch) that expects a `Readable` but I'm having a lot of issues getting something from node-fetch's response.body that will work. 

If I try to just pass in `response.body` I get an error about no `.pipe()` existing, which seems in line with MDN's documentation. 

So now I'm trying `Readable.fromWeb` but I get this error: `Argument of type ReadableStream<Uint8Array> is not assignable to parameter of type ReadableStream<any>` and the code doesn't work. Is there an alternative that I can use in Remix's node package? 

<img width="1258" alt="image" src="https://github.com/remix-run/remix/assets/2825063/3be33dc8-ab80-42ac-9e49-da55d8e94d6a">

I've tried referencing this issue: https://github.com/remix-run/remix/issues/3862, but no luck so far implementing the suggestions there. 

Thank you!

**Comment by timothymalcham** [comment]
If anyone comes across this, I just ended up using Axios for this specific part of my remix app since you can use `response.body.pipe()` and get a `NodeJS.ReadableStream` that is compatible with a lot of node libraries out there. 

`readableStreamVar as unknown as ReadableStream<any>` also didn't work, and I didn't bother to look into what it would take to implement the expected interface, which sounds like it could be a solution based on some of the responses to above-mentioned github issue (the examples didn't work for my case).

**Comment by Markyiptw** [comment]
Hi @timothymalcham, just want to say thanks that I am having similar issue with the `node:stream` streams not compatible with the web `ReadableStream` and `WritableStream`. After spending hours with `.wrap()`, `.fromWeb()`, `.toWeb()` etc, I come accross this discussion, checking #3862 , trying out the utils at `@remix-run/node` and my problem seems solved with `writeReadableStreamToWritable`.

For anyone who are curious, I am trying to pipe `request.body` into `busboy` so I can process files without `unstable_parseMultipartFormData`.

**Reply by guansss** [reply]
Thank you, this helps me a lot! I was trying to download a file and now it's done with `writeReadableStreamToWritable(res.body, fs.createWriteStream(destination))`.

**Reply by mrienstra** [reply]
`writeReadableStreamToWritable` source:
https://github.com/remix-run/remix/blob/main/packages/remix-node/stream.ts#L4-L29

In the (current) latest commit to `main` ("permalink", in case it moves in the future):
https://github.com/remix-run/remix/blob/58ac1e9/packages/remix-node/stream.ts#L4-L29
